src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -3593,14 +3593,26 @@ void AuraEffect::HandleModArmorPctFromStat(AuraApplication const* aurApp, uint8
     player->UpdateArmor();
 }
 
-void AuraEffect::HandleModStatBonusArmor(AuraApplication const* aurApp, uint8 mode, bool apply) const
+void AuraEffect::HandleModBonusArmor(AuraApplication const* aurApp, uint8 mode, bool apply) const
 {
     if (!(mode & (AURA_EFFECT_HANDLE_CHANGE_AMOUNT_MASK | AURA_EFFECT_HANDLE_STAT)))
         return;
 
     aurApp->GetTarget()->HandleStatFlatModifier(UNIT_MOD_ARMOR, BASE_VALUE, float(GetAmount()), apply);
 }
 
+void AuraEffect::HandleModBonusArmorPercent(AuraApplication const* aurApp, uint8 mode, bool /*apply*/) const
+{
+    if (!(mode & (AURA_EFFECT_HANDLE_CHANGE_AMOUNT_MASK | AURA_EFFECT_HANDLE_STAT)))
+        return;
+
+    Player* target = aurApp->GetTarget()->ToPlayer();
```

Comment:
target does not have to be a player, you can simply call `target->UpdateArmor()` on the `Unit*` object. Polymorphism magic does the rest

(not sure if this is relevant for creatures though)

Reply:
thanks :D